.github/workflows/main.yaml

```diff
@@ -0,0 +1,23 @@
+name: "Compile and Test"
+
+on:
+  push:
+    branches: [master]
+  pull_request:
+  workflow_dispatch:
+
+jobs:
+  all:
+    runs-on: ubuntu-22.04
+
+    steps:
+      - uses: actions/checkout@v4
+        name: "Clone the repository"
```

Comment:
TLDR: `action` is a reusable process on GitHub Actions. If someone already created the process and publish it to the marketplace, then I can reuse it.